src/main/java/net/snowflake/ingest/streaming/OpenChannelRequest.java

```diff
@@ -43,6 +43,10 @@ public enum OnErrorOption {
   private final String offsetToken;
   private final boolean isOffsetTokenProvided;
 
+  // If true, the channel will be dropped when it is closed after any pending data is fully
+  // committed.
+  private final boolean dropOnClose;
```

Comment:
The original reason I did it this way was because I thought Channel extends `AutoCloseable` and therefore, we cannot change the signature of `close` but I was mistaken - only the client extends it. Since users of this class has to call `close` explicitly anyways, I think your suggestions is the less-intrusive approach. There is still a corner case of someone using try-with using the Client and closing channels that way implicitly without any option to drop them but I think that's likely a rare corner case. We can always go and implement my idea if that is requested.